app/javascript/src/submit_handler.js

```diff
@@ -0,0 +1,108 @@
+
+const {
+  mergeObjects
+} = require('./utilities');
+
+class SubmitHandler {
```

Comment:
I think we should add a little description to explain this class. We both know what it's doing but I'm not convinced it would be all that obvious for any future developer, having just scanned through properties and methods. Also, I've seen you do a good job of explaining config options elsewhere, so would benefit from something similar here, before you consider it a final version, of course.